[wifi-login-pages] Added remember me checkbox #63

codesankalp opened the pull request:
Added remember_me checkbox with sessionStorage.

Closes #63

codesankalp commented:
@nemesisdesign, @niteshsinha17 Can you review it?
I tried to implement the sessionStorage when `remember_me` is unchecked and made it compatible with all other previous code.

niteshsinha17 commented:
@codesankalp You have formatted lots of files. It made it difficult to review also to understand which change is made for this particular issue.

codesankalp commented:
@niteshsinha17 I will unstage the reformatted file in which I haven't added a code for this implementation.

codesankalp commented:
@niteshsinha17 I have removed the extra reformatted files.

Also, I have implemented the remember_me feature only for the login.
If it's required for registration then do let me know.

codesankalp commented:
@nemesisdesign Can you please review it?
I am using commit of this PR in https://github.com/openwisp/openwisp-wifi-login-pages/issues/171.
And will submit a PR soon.

nemesisdesign commented:
@codesankalp sorry, I need more days.

nemesisdesign commented on client/components/login/login.js:
```diff
@@ -59,8 +60,8 @@ export default class Login extends React.Component {
-    const {email, errors} = this.state;
-    const {language} = this.props;
+    const { email, errors } = this.state;
+    const { language } = this.props;
```
please avoid these kind of style changes. If you want to add an automatic formatting tool that does this kind of change, please open a new issue and a new PR just for that.

This applies to any change in this PR which is not directly related to the feature you're working on.

nemesisdesign commented on internals/generators/organization/config.yml.hbs:
```diff
@@ -422,6 +422,10 @@ client:
+        remember_me:
+          type: "checkbox"
+          label:
+            en: "Remember Me"
```
let's change this to "Automatically log me in to the WiFi service at the next visit" which I hope most end-users will be able to understand

nemesisdesign commented on org-configurations/default-configuration.yml:
```diff
@@ -159,7 +159,7 @@ client:
             en: "only letters, numbers, and @/./+/-/_ characters"
           placeholder:
-            en: "enter location" 
+            en: "enter location"
```
avoid this unrelated change please

nemesisdesign commented on client/components/login/login.test.js:
```diff
@@ -110,6 +110,15 @@ describe("<Login /> interactions", () => {
+      .simulate("change", { target: { checked: false, name: "remember_me" } });
+    expect(wrapper.state("remember_me")).toEqual(false);
+  });
+
```
Good that you added one test, although looks not sufficient to prevent this feature from breaking.
Is there any measure you can include in this PR so that if someone changes the code and breaks this feature, the tests will find out?

nemesisdesign reviewed (changes requested):
Sorry for my delay @codesankalp, here's my review.

First of all, thank you for your effort, I tested it and it works! 
We need to improve it a bit, see below.

### Remember me choice should be maintained

If I choose to opt out from "remember me" and I close and then open the browser again, the checkbox should remain unchecked because otherwise the user could inadvertently log-in with the option activated from a shared computer. Can you please do this change (we can use localStorage)?

### Configurable default value

Can you make the default value of the checkbox configurable?
Some orgs may want to have this turned on by default while others off by default.

codesankalp commented:
Sorry for the delay @nemesisdesign Most of the work is done but needs some more time in adding tests for session storage so that this feature can be tested well and it will not break again in the future.

codesankalp commented:
@nemesisdesign Needs your review.

nemesisdesign commented on internals/generators/organization/index.js:
```diff
@@ -85,6 +85,12 @@ module.exports = {
+    {
+      type: "confirm",
+      name: "remember_me",
+      message: "Does your captive portal support automatically log me in?",
```
mm.. did I ask you to add this?

If I am not missing anything, this is not a feature of the captive portal but something we are adding on top, because we save the session of the user from which we're then able to retrieve the radius user token and log in the user automatically.

So the question should be something like: `Do you want to support automatic logins (remember me checkbox)?`

nemesisdesign commented on org-configurations/default-configuration.yml:
```diff
@@ -340,6 +340,11 @@ client:
+          type: "checkbox"
+          value: true
+          label:
+            en: "Automatically log me in to the WiFi service at the next visit"
```
after testing it I noticed it's too long, can you please shorten this and the other occurrences to something more conservative like: `Remember me on this device`?

nemesisdesign reviewed (changes requested):
Thank you @codesankalp, this is almost ready, see below for my final requests for improvement.

codesankalp commented:
@nemesisdesign Should I squash the commits?

nemesisdesign commented:
> @nemesisdesign Should I squash the commits?

No, don't worry, I'll do it.

nemesisdesign reviewed (approved):
Thanks! :+1:
